pacttesting/mock_service.go

```diff
@@ -90,6 +93,40 @@ func (m *MockServer) writePidFile() {
 	}
 }
 
+// Stop gracefully shuts does the underlying pact-mock-service process
+// and removes the server metadata (pid) file.
+func (m *MockServer) Stop() error {
+	p, err := os.FindProcess(m.Pid)
+	if err == nil {
+		err = p.Signal(os.Interrupt)
+		if err != nil {
+			return errors.WithMessage(err, fmt.Sprintf("failed to send interrupt to pid %d", m.Pid))
+		}
+
+		// wait for process to exit after interrupt, if it fails to stop
+		// then kill it
+		if err := retry.Do(func() error {
+			// check if the process is still alive
+			return p.Signal(syscall.Signal(0))
+		}, retry.Timeout(5*time.Second), retry.Sleep(200*time.Millisecond)); err != nil {
+			log.Printf("server failed to stop after interrupt, killing process...")
+			p.Kill()
```

Comment:
should we return an error in this case, since the file might not have been written out if the process had to be killed?

Reply:
I wondered about this, StopMockServers doesn't return an error so callers will never know if this failed or not. I think we should still return an error here though for consistency within this function.